pkg/router/template/template_helper.go

```diff
@@ -347,9 +351,20 @@ func clipHAProxyTimeoutValue(val string) string {
 			return haproxyMaxTimeout
 		}
 	} else {
-		duration, err := time.ParseDuration(val)
+		var duration, err = templateutil.ParseDurationWithErrors(val)
```

Comment:
Why change `:=` to `var`?

Why not add support for the "d" unit to `templateutil.ParseDurationWithErrors`? Then `clipHAProxyTimeoutValue` wouldn't need to special-case the `val[endIndex] == 'd'` case, and it would properly handle values like "1d1s".